tap-version-14-specification.md

```diff
@@ -0,0 +1,659 @@
+# need to ping 6 servers
+ok 2 - pinged diamond
+ok 3 - pinged ruby
+not ok 4 - pinged saphire
+  ---
+  message: 'hostname "saphire" unknown'
+  severity: fail
+  ...
+ok 5 - pinged onyx
+not ok 6 - pinged quartz
+  ---
+  message: 'timeout'
+  severity: fail
+  ...
+ok 7 - pinged gold
+1..7
+```
+
+### Giving up
+
+This listing reports that a pile of tests are going to be run. However, the
+first test fails, reportedly because a connection to the database could not
+be established. The program decided that continuing was pointless and
+exited.
+
+```tap
+TAP version 14
+1..573
+not ok 1 - database handle
+Bail out! Couldn't connect to database.
+```
+
+### Skipping a few
+
+The following listing plans on running 5 tests. However, our program
+decided to not run tests 2 thru 5 at all. To properly report this, the
+tests are marked as being skipped.
+
+```tap
+TAP version 14
+1..5
+ok 1 - approved operating system
+# $^0 is solaris
+ok 2 - # SKIP no /sys directory
+ok 3 - # SKIP no /sys directory
+ok 4 - # SKIP no /sys directory
+ok 5 - # SKIP no /sys directory
+```
+
+### Skipping everything
+
+This listing shows that the entire listing is a skip. No tests were run.
+
+```tap
+TAP version 14
+1..0 # skip because English-to-French translator isn't installed
+```
+
+## Got spare tuits?
+
+The following example reports that four tests are run and the last two
+tests failed. However, because the failing tests are marked as things to do
+later, they are considered successes. Thus, a harness should report this
+entire listing as a success.
+
+```tap
+TAP version 14
+1..4
+ok 1 - Creating test program
+ok 2 - Test program runs, no error
+not ok 3 - infinite loop # TODO halting problem unsolved
+not ok 4 - infinite loop 2 # TODO halting problem unsolved
+```
+
+## Creative liberties
+
+This listing shows an alternate output where the test numbers aren't
+provided. The test also reports the state of a ficticious board game as a
+YAML block. Finally, the test count is reported at the end.
+
+```tap
+TAP version 14
+ok - created Board
+ok
+ok
+ok
+ok
+ok
+ok
+ok
+  ---
+  message: "Board layout"
+  severity: comment
+  dump:
+     board:
+       - '      16G         05C        '
+       - '      G N C       C C G      '
+       - '        G           C  +     '
+       - '10C   01G         03C        '
+       - 'R N G G A G       C C C      '
+       - '  R     G           C  +     '
+       - '      01G   17C   00C        '
+       - '      G A G G N R R N R      '
+       - '        G     R     G        '
+  ...
+ok - board has 7 tiles + starter tile
+1..9
+```
+
+## BUGS
+
+Feature requests and bug reports should be [raised on
+GitHub.](https://github.com/TestAnything/testanything.github.io/issues/new)
+
+## AUTHORS
+
+The original TAP documentation (of which this is a hacked about version)
+was written by Andy Lester, based on the original Test::Harness
+documentation by Michael Schwern.  This version was written by Andy
```

Comment:
@isaacs, I think you could attach your name to this version.

Reply:
Haha, yeah, didn't get that far down yet, but I'll definitely update it with all who contribute to this version (myself included).

Reply:
fc4b81311e9b55baefda6be55ec2539f992695c4